src/resources/ble_esp32.cc

```diff
@@ -546,8 +533,7 @@ PRIMITIVE(init) {
 
   ble_hs_cfg.sync_cb = ble_on_sync;
 
-  group->register_resource(gap);
-  group->set_gap(gap);
+  nimble_port_init();
```

Comment:
Does the BLEEventSource thread start running (_should_run == true) at the BLEEventSource::instance() call? If so, It feels a bit weird (and potentially riddled with race conditions) that we're not done with the configuration and we haven't called init yet, but we still let it start.

Reply:
No. It starts when a GAP resource is registered. That is why the register_resource call is moved after the server init.

Reply:
Got it! Maybe add that as a comment just above the call to `group->register_resource(gap)`?